fog/view/enclave/impl/src/lib.rs

```diff
@@ -263,15 +264,20 @@ where
     fn create_client_query_response(
         &self,
         client_query_request: QueryRequest,
-        shard_query_responses: BTreeMap<ResponderId, EnclaveMessage<ClientSession>>,
+        shard_query_responses: BTreeMap<ResponderId, EnclaveMessage<NonceSession>>,
     ) -> Result<QueryResponse> {
-        let encrypted_shard_query_response = shard_query_responses
-            .values()
+        // Choose any shard query response and use it to supply the skeleton values for
+        // the QueryResponse. The tx_out_search_results and
+        // highest_processed_block_count fields will be set based on all of the
+        // shard query responses.
+        let shard_query_response = shard_query_responses
+            .clone()
+            .into_iter()
             .next()
-            .expect("Shard query responses must have at least one response.")
-            .clone();
-        let shard_query_response_plaintext =
-            self.ake.client_decrypt(encrypted_shard_query_response)?;
+            .expect("Shard query responses must have at least one response.");
```

Comment:
:wrench: Is the `clone()` necessary here?
(sorry github suggestion failure here)
```rust
        let shard_query_response = shard_query_responses
            .into_iter()
            .next()
            .expect("Shard query responses must have at least one response.");
```

Reply:
This ends up cloning the entire list in order to use the first element, but presumably `backend_decrypt()` is taking a reference? If so, what you want to do is something like this:

```
let plaintext = {
    let response = shard_query_responses
        .iter()
        .next()
        .expect("Shard query responses must have at least one response");
    self.ake.backend_decrypt(response.0, response.1)?
};
```

Reply:
Done.